src/components/PieChart.js

```diff
@@ -0,0 +1,182 @@
+const React = require('react');
+const ReactDOM = require('react-dom');
+const d3 = require('d3');
+
+const StyleConstants = require('../constants/Style');
+
+const PieChart = React.createClass({
+  propTypes: {
+    animateDuration: React.PropTypes.number,
+    animationTypeOnLoad: React.PropTypes.oneOf(['roll', 'pop']),
+    arcWidth: React.PropTypes.number,
+    colors: React.PropTypes.array,
+    data: React.PropTypes.array.isRequired,
+    height: React.PropTypes.number,
+    hoverExpandDistance: React.PropTypes.number,
+    onLabelClick: React.PropTypes.func,
+    onMouseEnter: React.PropTypes.func,
+    onMouseLeave: React.PropTypes.func,
+    onPieClick: React.PropTypes.func,
+    showDataLabel: React.PropTypes.bool,
+    style: React.PropTypes.object,
+    width: React.PropTypes.number
+  },
+
+  getDefaultProps () {
+    return {
+      animationTypeOnLoad: 'roll',
+      animateDuration: 500,
+      arcWidth: 2,
+      colors: [StyleConstants.Colors.PRIMARY].concat(d3.scale.category20().range()),
+      height: 350,
+      hoverExpandDistance: 10,
+      onLabelClick () {},
+      onMouseEnter () {},
+      onMouseLeave () {},
+      onPieClick () {},
+      showDataLabel: true,
+      width: 300
+    };
+  },
+
+  componentDidMount () {
+    const dom = ReactDOM.findDOMNode(this.refs.chart);
+
+    this._renderChart(dom);
+  },
```

Comment:
It will need to update when we use it in the Spending widget react rewrite as a user can dig into transactions and change transaction categories.  Something like this should work

```javascript
componentWillReceiveProps (nextProps) {
  if (!_isEgual(this.props.data, nextProps.data) {
     this._renderChart();
  }
}
```